### PR TITLE
Adds LOOC

### PR DESCRIPTION
**Pull request by UndeadNappist**
Adds LOOC. Works fine now.

### CODE REVIEW

File: code/modules/client/verbs/ooc.dm [hunk 1]
```diff
@@ -131,6 +131,106 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
...
+				if(holder) // final sanity check
+					display_name = key // admins display key as dead mobs.
+			to_chat(C, "<span class='looc'><span class='prefix'>LOOC:</span> <EM>[display_name]:</EM> <span class='message'>[msg]</span></span>")
+
+	for(var/client/C in GLOB.admins)
+		if(C.prefs.toggles & CHAT_LOOC)
+			var/prefix = "(R)LOOC"
+			if (C.mob in heard)
+				prefix = "LOOC"
+			to_chat(C, "<span class='looc'><span class='ooc'><span class='prefix'>[prefix]:</span> <EM>[src.key]:</EM> <span class='message'>[msg]</span></span>")
```
Comment (LetterN) [inline code/modules/client/verbs/ooc.dm]: `[src.mob.key]`

File: code/modules/client/verbs/ooc.dm [hunk 2]
```diff
@@ -131,6 +131,106 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
...
+			if(holder)
+				if(holder.fakekey)
+					if(C.holder)
+						display_name = "[holder.fakekey]/([key])"
+					else
+						display_name = holder.fakekey
+			if(istype(mob, /mob/dead/observer)) // admins
+				if(holder) // final sanity check
+					display_name = key // admins display key as dead mobs.
+			to_chat(C, "<span class='looc'><span class='prefix'>LOOC:</span> <EM>[display_name]:</EM> <span class='message'>[msg]</span></span>")
```
Comment (LetterN) [inline code/modules/client/verbs/ooc.dm]: `[src.mob.name]`
also comment out the
`
			var/display_name = key
			if(holder)
				if(holder.fakekey)
					if(C.holder)
						display_name = "[holder.fakekey]/([key])"
					else
						display_name = holder.fakekey
			if(istype(mob, /mob/dead/observer)) // admins
				if(holder) // final sanity check
					display_name = key // admins display key as dead mobs.
`

File: code/modules/client/verbs/ooc.dm [hunk 3]
```diff
@@ -131,6 +131,108 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
...
+					display_name = key // admins display key as dead mobs.
+*/
+			to_chat(C, "<span class='looc'><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
+
+	for(var/client/C in GLOB.admins)
+		if(C.prefs.toggles & CHAT_LOOC)
+			var/prefix = "(R)LOOC"
+			if (C.mob in heard)
+				prefix = "LOOC"
+			to_chat(C, "<span class='looc'><span class='ooc'><span class='prefix'>[prefix]:</span> <EM>[src.mob.key]:</EM> <span class='message'>[msg]</span></span>")
```
Comment (LetterN) [inline code/modules/client/verbs/ooc.dm]: <EM>[src.key]/[src.mob.name]:</EM>
whoops, forgot about that

Comment (LetterN) [reply code/modules/client/verbs/ooc.dm]: in the <em>

**Comment by UndeadNappist** [comment]
LOOC now works. Thanks to whoever pointed me at yog's code, ThatLing, and Super322.

File: code/modules/client/verbs/ooc.dm [hunk 4]
```diff
@@ -131,6 +131,148 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
 		prefs.ooccolor = initial(prefs.ooccolor)
 		prefs.save_preferences()
 
+/client/verb/looc(msg as text)
```
Comment (LetterN) [inline code/modules/client/verbs/ooc.dm]: preferably if looc is on it's own file next to ooc

File: code/modules/client/verbs/looc.dm [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+			return
+	else
+		GLOB.dlooc_allowed = !GLOB.dlooc_allowed
+
+/client/proc/get_looc()
+	var/msg = input(src, null, "looc \"text\"") as text|null
+	looc(msg)
+
+//Checks admin notice
+/client/verb/admin_notice()
```
Comment (LetterN) [inline code/modules/client/verbs/looc.dm]: why did this move??

Comment (UndeadNappist) [reply code/modules/client/verbs/looc.dm]: Finger slipped.

**Pull request merged**